program_base.go

```diff
@@ -121,15 +122,16 @@ type BaseProgram struct {
 
 // Load loads program into linux kernel
 func (prog *BaseProgram) Load() error {
-	// Sanity checks
-	if len(prog.name) >= C.BPF_OBJ_NAME_LEN {
-		return fmt.Errorf("Program name '%s' is too long", prog.name)
+
+	// truncate name if required
+	pname := prog.name
+	if len(pname) >= C.BPF_OBJ_NAME_LEN {
+		pname = pname[:int(C.BPF_OBJ_NAME_LEN)]
 	}
```

Comment:
Personally I'd prefer to throw error instead of silently truncate name unless there is any major reason not to do that

Reply:
I used this when experimenting with the program name being used as the kprobe point, will change it back to returning an error instead.